modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/CommonScriptMessageContext.java

```diff
@@ -0,0 +1,848 @@
+    private static final String JSON_TEXT = "JSON_TEXT";
+
+    /** The actual Synapse message context reference */
+    private final MessageContext mc;
+    /** The OMElement to scripting language object converter for the selected language */
+    private final XMLHelper xmlHelper;
+
+
+    private ScriptEngine scriptEngine;
+
+    public CommonScriptMessageContext(MessageContext mc, XMLHelper xmlHelper) {
+        this.mc = mc;
+        this.xmlHelper = xmlHelper;
+    }
+
+    /**
+     * Get the XML representation of SOAP Body payload.
+     * The payload is the first element inside the SOAP <Body> tags
+     *
+     * @return the XML SOAP Body
+     * @throws ScriptException in-case of an error in getting
+     * the XML representation of SOAP Body payload
+     */
+    @Override
+    public Object getPayloadXML() throws ScriptException {
+        return xmlHelper.toScriptXML(mc.getEnvelope().getBody().getFirstElement());
+    }
+
+    /**
+     * Set the SOAP body payload from XML.
+     *
+     * @param payload Message payload
+     * @throws ScriptException For errors in converting xml To OM
+     * @throws OMException     For errors in OM manipulation
+     */
+
+    @Override
+    public void setPayloadXML(Object payload) throws OMException, ScriptException {
+        SOAPBody body = mc.getEnvelope().getBody();
+        OMElement firstChild = body.getFirstElement();
+        OMElement omElement = xmlHelper.toOMElement(payload);
+        if (firstChild == null) {
+            body.addChild(omElement);
+        } else {
+            firstChild.insertSiblingAfter(omElement);
+            firstChild.detach();
+        }
+    }
+
+    /**
+     * Get the JSON object representation of the JSON message body of the request.
+     *
+     * @return JSON object of the message body
+     */
+    @Override
+    public Object getPayloadJSON() {
+        return jsonObject(mc);
+    }
+
+    @Override
+    public Object getJsonText() {
+        if (mc == null) {
+            return "";
+        }
+        Object text = mc.getProperty(JSON_TEXT);
+        return text == null ? "{}" : text;
+    }
+
+    /**
+     * Get the Message Payload as a text.
+     *
+     * @return Payload as text
+     */
+    @Override
+    public String getPayloadText() {
+        if (JsonUtil.hasAJsonPayload(((Axis2MessageContext) mc).getAxis2MessageContext())) {
+            return JsonUtil.jsonPayloadToString(((Axis2MessageContext) mc).getAxis2MessageContext());
+        } else {
+            return mc.getEnvelope().toString();
+        }
+    }
+
+    /**
+
+    /**
+     * Saves the payload of this message context as a JSON payload.
+     *
+     * @param jsonPayload Javascript native object to be set as the message body
+     * @throws ScriptException in case of creating a JSON object out of
+     *                         the javascript native object.
+     */
+    public void setPayloadJSON0(Object jsonPayload) throws ScriptException {
```

Comment:
Hi I have only renamed the existing synapse ScriptMessageContext class to  CommonScriptMessageContext. Since these are existing public methods there may be users who are using these method names in scripts.